code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -753,280 +753,176 @@
+		playsound(get_turf(buffing_target), "alien_roar", 40)
+		buffing_target.create_custom_empower(icolor = "#a31010", ialpha = 200, small_xeno = TRUE)
+		buffing_target.add_filter("raging", 1, list("type" = "outline", "color" = "#a31010", "size" = 1))
+		buffing_target.armor_modifier += target_armor_buff
+		buffing_target.recalculate_armor()
 
-		to_chat(wardenbuff_user, SPAN_XENOHIGHDANGER("We rejuvenate [targetXeno]!"))
-		to_chat(targetXeno, SPAN_XENOHIGHDANGER("We are rejuvenated by [wardenbuff_user]!"))
-		targetXeno.visible_message(SPAN_BOLDNOTICE("[wardenbuff_user] points at [targetXeno], and it spasms as it recuperates unnaturally quickly!")) //marines probably should know if a xeno gets rejuvenated
-		targetXeno.xeno_jitter(1 SECONDS) //it might confuse them as to why the queen got up half a second after being AT rocketed, and give them feedback on the Praetorian rejuvenating
-		targetXeno.flick_heal_overlay(3 SECONDS, "#F5007A") //therefore making the Praetorian a priority target
-		targetXeno.clear_debuffs()
-		use_plasma = TRUE
-	if (use_plasma)
-		use_plasma_owner()
+	addtimer(CALLBACK(src, PROC_REF(remove_rage)), armor_buffs_duration)
+	addtimer(CALLBACK(src, PROC_REF(remove_target_rage)), armor_buffs_targer_dur)
 
 	apply_cooldown()
 	return ..()
 
-/datum/action/xeno_action/activable/prae_retrieve/use_ability(atom/A)
-	var/mob/living/carbon/xenomorph/warden = owner
-	if(!istype(warden))
-		return
 
-	var/datum/behavior_delegate/praetorian_warden/behavior = warden.behavior_delegate
-	if(!istype(behavior))
-		return
+/datum/action/xeno_action/activable/valkyrie_rage/proc/remove_rage()
+	var/mob/living/carbon/xenomorph/raging_valkyrie = owner
+	var/datum/behavior_delegate/praetorian_valkyrie/behavior = raging_valkyrie.behavior_delegate
+	raging_valkyrie.remove_filter("raging")
+	raging_valkyrie.armor_modifier -= armor_buff
+	armor_buffs_active = FALSE
+	behavior.raging = FALSE
+	raging_valkyrie.recalculate_armor()
+	to_chat(raging_valkyrie, SPAN_XENOHIGHDANGER("We feel ourselves calm down."))
 
-	if(warden.observed_xeno != null)
-		to_chat(warden, SPAN_XENOHIGHDANGER("We cannot retrieve sisters through overwatch!"))
-		return
 
-	if(!isxeno(A) || !warden.can_not_harm(A))
-		to_chat(warden, SPAN_XENODANGER("We must target one of our sisters!"))
-		return
 
-	if(A == warden)
-		to_chat(warden, SPAN_XENODANGER("We cannot retrieve ourself!"))
-		return
 
-	if(!(A in view(7, warden)))
-		to_chat(warden, SPAN_XENODANGER("That sister is too far away!"))
-		return
+/datum/action/xeno_action/activable/valkyrie_rage/proc/remove_target_rage()
+	var/mob/living/carbon/xenomorph/target_xeno = focus_rage.resolve()
+	if(target_xeno) //if the target was qdeleted it would be null so you need to check for it
+		target_xeno.armor_modifier -= target_armor_buff
+		target_xeno.remove_filter("raging")
+		armor_buffs_active_target = FALSE
+		target_xeno.recalculate_armor()
+		to_chat(target_xeno, SPAN_XENOHIGHDANGER("We feel ourselves calm down."))
```

Comment:
```suggestion
	if(target_xeno) //if the target was qdeleted it would be null so you need to check for it
		target_xeno.armor_modifier -= target_armor_buff
		target_xeno.remove_filter("raging")
		target_xeno.recalculate_armor()
		to_chat(target_xeno, SPAN_XENOHIGHDANGER("We feel ourselves calm down."))
	armor_buffs_active_target = FALSE
```
should be set on false regardless of the target being qdeleted or not